src/core/api/ldap.go

```diff
@@ -51,6 +54,19 @@ func (l *LdapAPI) Prepare() {
 		return
 	}
 
+	// check the auth_mode except ping
+	if strings.EqualFold(l.Ctx.Request.RequestURI, "/api/ldap/ping") {
+		return
+	}
+	authMode, err := config.AuthMode()
+	if err != nil {
+		l.HandleInternalServerError(fmt.Sprintf("Can't load system configuration, error: %v", err))
+		return
+	}
+	if authMode != "ldap_auth" {
+		l.HandleInternalServerError(fmt.Sprintf("current auth_mode is not ldap_auth"))
```

Comment:
Have you tested this?  I'm afraid this doesn't work because the `return` will only return from `Prepare` and the request will be handled.

Reply:
Verified, HandlerInternalServerError works in Prepare